timeline-chart/src/layer/time-graph-chart.ts

```diff
@@ -487,7 +492,13 @@ export class TimeGraphChart extends TimeGraphChartLayer {
                     this.ongoingRequest = undefined;
                 }
                 this.isNavigating = false;
+                if (this._coarseResolutionFactor !== FINE_RESOLUTION_FACTOR) {
```

Comment:
I realized that if this is already the 2nd (fine) update, we will unnecessarily debounce a 3rd fine update. Although when it executes, it is likely to have no rows to update and abort. But we still should not attempt it if it was already the fine update.